File: test_cases/javascript/default_parameters_added/after.js
// After: Functions with default parameters added
function greet(name = 'World', greeting = 'Hello') {
    return `${greeting}, ${name}!`;
}

function createUser(name, age = 18, role = 'user', active = true) {
    return { name, age, role, active };
}

function processArray(arr, mapper = x => x, filter = () => true) {
    return arr.filter(filter).map(mapper);
}

class Calculator {
    constructor(precision = 2) {
        this.precision = precision;
    }
    
    format(value, decimals = this.precision) {
        return value.toFixed(decimals);
    }
    
    calculate(a, b, operation = 'add', round = true) {
        let result;
        switch (operation) {
            case 'add': result = a + b; break;
            case 'multiply': result = a * b; break;
            default: result = 0;
        }
        return round ? Math.round(result) : result;
    }
}

const processData = (data, options = { validate: true, transform: false }) => {
    return data;
};

const fetchData = async (url, timeout = 5000, retries = 3) => {
    // fetch with timeout and retries
    return fetch(url);
};

File: test_cases/python/default_parameters_added/after.py
def greet(name, greeting="Hello", punctuation="!"):
    """Function with default parameters."""
    return f"{greeting}, {name}{punctuation}"

def calculate(x, y=1, z=0):
    """Simple calculation function with defaults."""
    return x + y * z

def process_data(data, options=None, debug=False):
    """Data processing function with defaults."""
    if options is None:
        options = []
    
    if debug:
        print(f"Processing {len(data)} items")
    
    return [item for item in data if item in options]
